Assets/Scripts/HexTile.cs

```diff
@@ -12,18 +12,36 @@ public class HexTile : MonoBehaviour
     // Start is called before the first frame update
     void Start()
     {
+        // Check whether a MeshFilter or MeshCollider exist on the Hex. If at least one of them does, then generate the
+        // mesh.
         MeshFilter meshFilter = gameObject.GetComponent<MeshFilter>();
+        MeshCollider meshCollider = gameObject.GetComponent<MeshCollider>();
+        if (meshFilter || meshCollider)
+        {
+            mesh = GenerateMesh();
+        }
+        
+        // If the MeshFilter exists, assign the generated mesh.
         if (!meshFilter)
         {
             Debug.LogWarning("HexTile at position " + pos + " has no MeshFilter. It will not be visible in " +
                              "the scene.");
         }
         else
         {
-            Debug.Log("Mesh filter found. Generating mesh.");
-            mesh = GenerateMesh();
             meshFilter.mesh = mesh;
         }
+
+        // If the MeshCollider exists, assign the generated mesh.
+        if (!meshCollider)
+        {
+            Debug.Log("HexTile at position " + pos + " has no MeshCollider.");
+        }
+        else
+        {
+            Debug.Log("MeshCollider found, adding Mesh.");

```

Comment:
The warning log for MeshFilter makes sense because I think most of the time the Hex should be visible. Colliders aren't as necessary, so these log lines may clutter things up. Consider adding a debugging check to decide whether or not to log.

Reply:
Removed both debug logs.